src/java/org/apache/cassandra/cql3/statements/UpdateStatement.java

```diff
@@ -348,4 +371,60 @@ public AuditLogContext getAuditLogContext()
     {
         return new AuditLogContext(AuditLogEntryType.UPDATE, keyspace(), table());
     }
+
+    public static void evaluateConstraints(TableMetadata tableMetadata, List<ColumnIdentifier> columnNames, List<Term.Raw> columnValues)
+    {
+        Map<String, Term.Raw> columnMap = getColumnMap(columnNames, columnValues);
+        for (ColumnMetadata column : tableMetadata.columns())
+        {
+            if (column.hasConstraint())
+                for (ColumnConstraint constraint : column.getColumnConstraints().getConstraints())
+                    evaluateConstraint(column, constraint, columnMap, column.type.getSerializer(), column);
+        }
+    }
+
+    public static void evaluateConstraint(ColumnMetadata column, ColumnConstraint constraint, Map<String, Term.Raw> columnMap, TypeSerializer serializer, ColumnSpecification receiver)
+    {
+        constraint.evaluate(serializer.deserialize(columnMap.get(column.name.toCQLString())
+                                                            .prepare(column.ksName, receiver).bindAndGet(QueryOptions.DEFAULT)));
+    }
+
+    public static void evaluateJsonConstraints(TableMetadata tableMetadata, List<ColumnIdentifier> columnNames, List<Term.Raw> columnValues)
+    {
+        Map<String, Term.Raw> columnMap = getColumnMap(columnNames, columnValues);
+        for (ColumnMetadata column : tableMetadata.columns())
+        {
+            if (column.hasConstraint())
+            {
+                // This is needed becouse when parsing from JSON, the literal type is for numeric
+                // values is float.
```

Comment:
nit: typo

```suggestion
                // This is needed because when parsing from JSON, the literal type for numeric
                // values is float.
```